pkg/metrics/metrics.go

```diff
@@ -0,0 +1,164 @@
+package metrics
+
+import (
+	"context"
+	"github.com/kubeslice/kubeslice-monitoring/pkg/util"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/runtime"
+	"time"
+)
+
+// MetricRecorder provides a container with config parameters for the Prometheus Exporter
+type MetricRecorder struct {
+	Logger    *zap.SugaredLogger
+	Scheme    *runtime.Scheme
+	Version   string
+	Project   string
+	Cluster   string
+	Slice     string
+	Namespace string
+	Subsystem string
+}
+
+type Metric struct {
+	Type             MetricType
+	Name             string
+	Help             string
+	Value            float64
+	Labels           map[string]string
+	histogramBuckets []float64
+}
+
+type MetricType string
+
+var fixedLabels = []string{"slice_project", "slice_cluster", "slice_name"}
+
+const (
+	MetricTypeGauge     = "Gauge"
+	MetricTypeCounter   = "Counter"
+	MetricTypeHistogram = "Histogram"
+	MetricTypeSummary   = "Summary"
+)
+
+// NewPrometheusProvider returns a Provider that produces Prometheus metrics.
+// Namespace and subsystem are applied to all produced metrics.
+func (mr *MetricRecorder) NewPrometheusProvider() *MetricRecorder {
+	return &MetricRecorder{
+		Logger:    mr.Logger,
```

Comment:
It looks like this method is to create a copy of an existing MetricProvider object. If that is the intention, the method name should be `Copy`

Reply:
Correct. Changing.